src_c/draw.c

```diff
@@ -264,6 +271,13 @@ aalines(PyObject *self, PyObject *arg, PyObject *kwargs)
                             surf->format->BytesPerPixel);
     }
 
+    if (!blend) {
+        PyErr_WarnEx(PyExc_DeprecationWarning,
```

Comment:
Missing check of the return value from `PyErr_WarnEx`  If it returns `-1` then an exception was thrown ( see https://docs.python.org/3/c-api/exceptions.html#issuing-warnings), which in turn requires an early return from this function.